tests/hw-accel/amdgpu/basic/tests/basic-test.go

```diff
@@ -0,0 +1,121 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/amdgpu"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"github.com/openshift-kni/eco-gotests/tests/hw-accel/amdgpu/internal/deviceconfig"
+	"github.com/openshift-kni/eco-gotests/tests/hw-accel/amdgpu/internal/labels"
+	"github.com/openshift-kni/eco-gotests/tests/hw-accel/amdgpu/internal/pods"
+	amdparams "github.com/openshift-kni/eco-gotests/tests/hw-accel/amdgpu/params"
+	"github.com/openshift-kni/eco-gotests/tests/internal/inittools"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("AMD GPU Basic Tests", Ordered, Label(amdparams.LabelSuite), func() {
+
+	Context("AMD GPU Basic 01", Label(amdparams.LabelSuite+"-01"), func() {
+
+		apiClient := inittools.APIClient
+
+		amdListOptions := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", amdparams.AMDNFDLabelKey, amdparams.AMDNFDLabelValue),
+		}
+
+		amdNodeBuilders, amdNodeBuildersErr := nodes.List(apiClient, amdListOptions)
+
+		BeforeAll(func() {
+
+			Expect(amdNodeBuildersErr).To(BeNil(),
+				fmt.Sprintf("Failed to get Builders for AMD GPU Worker Nodes. Error:\n%v\n", amdNodeBuildersErr))
+
+			Expect(amdNodeBuilders).ToNot(BeEmpty(),
+				"'amdNodeBuilders' can't be empty")
+
+		})
+
+		It("Check AMD label was added by NFD", func() {
+
+			// The assumption is that all Worker Nodes are equipped with AMD GPUs.
+			amdNFDLabelFound, amdNFDLabelFoundErr := labels.LabelPresentOnAllNodes(
+				apiClient, amdparams.AMDNFDLabelKey, amdparams.AMDNFDLabelValue, inittools.GeneralConfig.WorkerLabelMap)
+
+			Expect(amdNFDLabelFoundErr).To(BeNil(),
+				"An error occurred while attempting to verify the AMD label by NFD: %v ", amdNFDLabelFoundErr)
+			Expect(amdNFDLabelFound).To(BeTrue(),
+				"AMD label check failed to match label %s and label value %s on all nodes",
+				amdparams.AMDNFDLabelKey, amdparams.AMDNFDLabelValue)
+		})
+
+		It("Node Labeller", func() {
+
```

Comment:
Same here, for each It() testcase, add a a By(.... description of testcase...) step or a glog statement to explain what is tested, so in logs you can see what each testcase is doing